File: PerfTest2/PerfTest2-Xamarin/PerfTest2-Xamarin.Shared/Utilities/FileUtilities.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace PerfTest2Xamarin.Utilities
{
	public class FileUtilities : IDisposable
	{
		private string filePath;
		private StreamWriter streamWriter;

		private StreamWriter Writer => streamWriter ?? (streamWriter = new StreamWriter (filePath));

		public FileUtilities (string filePath)
		{
			filePath = Path.Combine (filePath, "testFile.txt");
		}

		public void CloseFile ()
		{
			if (streamWriter != null)
			{
				streamWriter.Close ();
				streamWriter.Dispose ();
				streamWriter = null;
			}
		}

		public void DeleteFile ()
		{
			if (File.Exists (filePath))
			{
				File.Delete (filePath);
			}
		}

		public void CreateFile ()
		{
			if (!File.Exists (filePath))
			{
				using (var stream = File.Create (filePath)) { }
			}
		}

		public void WriteLineToFile (String line)
		{
			if (!File.Exists (filePath))
				CreateFile ();

			Writer.WriteLine (line);
		}

		public IList<string> ReadFileContents ()
		{
			if (!File.Exists (filePath))
				return new List<string> ();

			using (var streamReader = new StreamReader (filePath))
			{
				var returnValue = new List<String> ();

				while (!streamReader.EndOfStream)
				{
					var line = streamReader.ReadLine ();
					if (line != null)
					{
						returnValue.Add (line);
					}
				}

				return returnValue;
			}
		}

		public void Dispose ()
		{
			CloseFile ();
		}
	}
}

File: PerfTest2/PerfTest2-Java/app/src/main/java/com/tamasszadvari/perftest2_java/FileUtilities.java
package com.tamasszadvari.perftest2_java;

import android.content.Context;
import android.os.Environment;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.util.ArrayList;

public class FileUtilities {
    private BufferedWriter fileHandle;
    private Context context;
    private File textFile;

    private BufferedWriter getFileHandle () throws Exception {
        if (fileHandle == null) {
            fileHandle = new BufferedWriter(new FileWriter(textFile));
        }

        return fileHandle;
    }

    public FileUtilities(Context context) {
        this.context = context;
        String filePath = context.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS) + File.separator + "testFile.txt";
        textFile = new File(filePath);
    }

    public void closeFile() throws Exception {
        if (fileHandle != null) {
            fileHandle.close();
            fileHandle = null;
        }
    }

    public void deleteFile() {
        if (textFile.exists()) {
            textFile.delete();
        }
    }

    public void createFile() throws Exception {
        if (!textFile.exists()) {
            textFile.createNewFile();
        }
    }

    public void writeLineToFile(String line) throws Exception {
        if (!textFile.exists()) {
            textFile.createNewFile();
        }

        getFileHandle().write(line);
    }

    public ArrayList<String> readFileContents() throws Exception {
        BufferedReader reader = new BufferedReader(new FileReader(textFile));
        ArrayList<String> returnValue = new ArrayList<>();
        String line;

        while((line = reader.readLine()) != null){
            returnValue.add(line);
        }

        reader.close();

        return returnValue;
    }
}